src/node/db/SecurityManager.js

```
@@ -57,86 +57,96 @@ exports.checkAccess = async (padID, sessionCookie, token, userSettings) => {
-    return DENY;
-  }
 
-  const sessionAuthorID = await p_sessionAuthorID;
-  if (settings.requireSession && !sessionAuthorID) {
-    authLogger.debug('access denied: HTTP API session is required');
-    return DENY;
-  }
+    if (!padID.includes('$')) {
+      // Only group pads can be private, so there is nothing more to check for this non-group pad.
+      return grant;
+    }
 
-  const grant = {
-    accessStatus: 'grant',
-    authorID: (sessionAuthorID != null) ? sessionAuthorID : await p_tokenAuthorID,
-  };
+    if (!padExists) {
+      if (sessionAuthorID == null) {
+        authLogger.debug('access denied: must have an HTTP API session to create a group pad');
+        return DENY;
+      }
+      // Creating a group pad, so there is no public status to check.
+      return grant;
+    }
 
-  if (!padID.includes('$')) {
-    // Only group pads can be private, so there is nothing more to check for this non-group pad.
-    return grant;
-  }
+    const pad = await padManager.getPad(padID);
 
-  if (!padExists) {
-    if (sessionAuthorID == null) {
-      authLogger.debug('access denied: must have an HTTP API session to create a group pad');
+    if (!pad.getPublicStatus() && sessionAuthorID == null) {
+      authLogger.debug('access denied: must have an HTTP API session to access private group pads');
       return DENY;
     }
-    // Creating a group pad, so there is no public status to check.
-    return grant;
-  }
-
-  const pad = await padManager.getPad(padID);
 
-  if (!pad.getPublicStatus() && sessionAuthorID == null) {
-    authLogger.debug('access denied: must have an HTTP API session to access private group pads');
+    return grant;
+  } catch (err) {
+    if (err.message === 'Query inactivity timeout') {
+      authLogger.debug('access denied: a necessary database query has timed out.');
+    } else {
+      authLogger.debug('access denied: an error occurred', err.message);
+    }
     return DENY;
   }
```

I don't think this is a good idea. If something throws, it would be better to let the error propagate, ultimately resulting in a 500 error to the user. Treating internal errors as unauthorized requests could have unintended side effects, such as triggering automatic firewall rules.

If you want to keep the debug logging I suggest doing this:

```suggestion
  } catch (err) {
    authLogger.debug(err.stack || err.toString());
    throw err;
  }
```

---

Good point re semantic difference between 403-like responses and 500. I'll fix the callers then.

---

PR #5171 should fix the biggest caller.

---

And now the `/ro/:id` handler is gone: 58bd96ce8f1e054e10e769506660ccb103c22a44